[0m2021.10.20 12:43:30 INFO  Started: Metals version 0.10.7 in workspace 'C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo' for client Visual Studio Code 1.60.2.[0m
[0m2021.10.20 12:43:31 INFO  time: initialize in 1.48s[0m
[0m2021.10.20 12:43:32 WARN  Build server is not auto-connectable.[0m
[0m2021.10.20 12:43:33 WARN  no build target for: C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo\build.sbt[0m
[0m2021.10.20 12:43:33 WARN  no build target for: C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo\src\main\scala\HelloWorld.scala[0m
Oct 20, 2021 12:43:38 PM org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint notify
INFO: Unsupported notification method: $/setTrace
[0m2021.10.20 12:43:38 INFO  running 'C:\Program Files\Java\jdk1.8.0_301\bin\java -Djline.terminal=jline.UnsupportedTerminal -Dsbt.log.noformat=true -Dfile.encoding=UTF-8 -jar C:\Users\WILLIA~1\AppData\Local\Temp\metals3238244063100422842\sbt-launch.jar -Dbloop.export-jar-classifiers=sources bloopInstall'[0m
[0m2021.10.20 12:43:41 INFO  no build target: using presentation compiler with only scala-library: 2.12.15[0m
[0m2021.10.20 12:43:45 INFO  [info] welcome to sbt 1.5.5 (Oracle Corporation Java 1.8.0_301)[0m
[0m2021.10.20 12:43:48 INFO  time: code lens generation in 11s[0m
[0m2021.10.20 12:43:48 INFO  time: code lens generation in 13s[0m
[0m2021.10.20 12:43:50 INFO  [info] loading settings for project scaladbdemo-build-build from metals.sbt ...[0m
[0m2021.10.20 12:43:51 INFO  [info] loading project definition from C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo\project\project[0m
Oct 20, 2021 12:43:53 PM org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint notify
INFO: Unsupported notification method: $/setTrace
[0m2021.10.20 12:43:56 INFO  [info] loading settings for project scaladbdemo-build from metals.sbt ...[0m
[0m2021.10.20 12:43:56 INFO  [info] loading project definition from C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo\project[0m
[0m2021.10.20 12:44:00 INFO  [success] Generated .bloop\scaladbdemo-build.json[0m
[0m2021.10.20 12:44:00 INFO  [success] Total time: 4 s, completed Oct 20, 2021 12:44:00 PM[0m
[0m2021.10.20 12:44:03 INFO  [info] loading settings for project scaladbdemo from build.sbt ...[0m
[0m2021.10.20 12:44:03 INFO  [info] set current project to HelloWorldApp (in build file:/C:/Users/William%20Terry/Desktop/BigDataOctober2021/ScalaDBDemo/)[0m
[0m2021.10.20 12:44:12 INFO  [success] Generated .bloop\scaladbdemo.json[0m
[0m2021.10.20 12:44:12 INFO  [success] Generated .bloop\scaladbdemo-test.json[0m
[0m2021.10.20 12:44:12 INFO  [success] Total time: 8 s, completed Oct 20, 2021 12:44:12 PM[0m
[0m2021.10.20 12:44:12 INFO  time: ran 'sbt bloopInstall' in 33s[0m
[0m2021.10.20 12:44:12 INFO  Attempting to connect to the build server...[0m
Starting the bsp launcher for bloop...
Opening a bsp server connection with 'bsp --protocol tcp --port 25299'...
Waiting for the bsp connection to come up...
[0m[32m[D][0m Waiting for a connection at 127.0.0.1:25299...
The server is listening for incoming connections at tcp://127.0.0.1:25299...
Starting thread that pumps stdin and redirects it to the bsp server...
Starting thread that pumps server stdout and redirects it to the client stdout...
[0m2021.10.20 12:44:13 INFO  tracing is disabled for protocol BSP, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\bsp.trace.json[0m
[0m2021.10.20 12:44:13 INFO  Attempting to connect to the build server...[0m
Starting the bsp launcher for bloop...
Opening a bsp server connection with 'bsp --protocol tcp --port 37694'...
Waiting for the bsp connection to come up...
[0m[32m[D][0m Waiting for a connection at 127.0.0.1:37694...
The server is listening for incoming connections at tcp://127.0.0.1:37694...
Starting thread that pumps stdin and redirects it to the bsp server...
Starting thread that pumps server stdout and redirects it to the client stdout...
[0m2021.10.20 12:44:14 INFO  tracing is disabled for protocol BSP, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\bsp.trace.json[0m
[0m2021.10.20 12:44:14 INFO  time: Connected to build server in 1.38s[0m
[0m2021.10.20 12:44:14 INFO  Connected to Build server: Bloop v1.4.9[0m
[0m2021.10.20 12:44:14 INFO  time: Imported build in 0.11s[0m
[0m2021.10.20 12:44:24 INFO  time: indexed workspace in 10s[0m
[0m2021.10.20 12:44:24 INFO  compiling scaladbdemo (1 scala source)[0m
[0m2021.10.20 12:44:25 INFO  no build target: using presentation compiler with only scala-library: 2.12.14[0m
[0m2021.10.20 12:44:51 INFO  time: compiled scaladbdemo in 27s[0m
[0m2021.10.20 12:45:37 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 12:45:37 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 12:45:38 INFO  Starting debug proxy for [Hello_World][0m
[0m2021.10.20 12:45:38 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:62670 .[0m
[0m2021.10.20 12:45:38 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 12:45:38 INFO  Canceling debug proxy for [Hello_World][0m
[0m2021.10.20 12:45:38 INFO  Closing debug server tcp://0.0.0.0:62667[0m
Oct 20, 2021 12:46:20 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation
WARNING: Unmatched cancel notification for request id 18
Oct 20, 2021 12:46:22 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation
WARNING: Unmatched cancel notification for request id 24
Oct 20, 2021 12:46:25 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation
WARNING: Unmatched cancel notification for request id 30
[0m2021.10.20 12:51:55 INFO  compiling scaladbdemo (4 scala sources)[0m
[0m2021.10.20 12:51:57 INFO  shutting down Metals[0m
[0m2021.10.20 12:51:57 INFO  Shut down connection with build server.[0m
[0m2021.10.20 12:51:57 INFO  Shut down connection with build server.[0m
No more data in the client stdin, exiting...
[0m2021.10.20 12:53:35 INFO  Started: Metals version 0.10.7 in workspace 'C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo' for client Visual Studio Code 1.60.2.[0m
[0m2021.10.20 12:53:35 INFO  time: initialize in 0.78s[0m
[0m2021.10.20 12:53:36 INFO  Attempting to connect to the build server...[0m
Oct 20, 2021 12:53:36 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleNotification
WARNING: Notification threw an exception: {
  "jsonrpc": "2.0",
  "method": "textDocument/didOpen",
  "params": {
    "textDocument": {
      "uri": "file:///c%3A/Users/William%20Terry/Desktop/BigDataOctober2021/ScalaDBDemo/src/main/scala/HelloWorld.scala",
      "languageId": "scala",
      "version": 1,
      "text": "import java.sql.DriverManager\r\nimport java.sql.Connection\r\nimport java.sql.SQLException\r\nimport java.sql.SQLTimeoutException\r\nimport java.sql.PreparedStatement\r\nimport java.io.FileReader\r\nimport java.io.BufferedReader\r\nimport java.io.FileNotFoundException\r\nimport scala.io.StdIn\r\n\r\nobject App {\r\n\r\n  val username \u003d \"admin\"\r\n  val password \u003d \"password\"\r\n\r\n  var connection:Connection \u003d null\r\n\r\n  def main(args: Array[String]): Unit \u003d {\r\n    //This application is a demo so the data is loaded into the database before user input is allowed for the simulation.\r\n\r\n    //Test Database connection\r\n    val connection_status \u003d test_mysql_connection()\r\n    if(connection_status \u003d\u003d 1) {\r\n      //Continue with program\r\n\r\n      //Load the data if empty\r\n      //load_books_to_db()\r\n\r\n      //Test connection with a simple query\r\n      //queryForBooksTest()\r\n\r\n      var finished \u003d false\r\n\r\n      LibraryPrompter.printLoginPrompt()\r\n      do {\r\n        SpecialPrint.cprint(\"username\u003e: \", Console.CYAN)\r\n        var input1 \u003d StdIn.readLine()\r\n        SpecialPrint.cprint(\"password\u003e: \", Console.CYAN)\r\n        var input2 \u003d StdIn.readLine()\r\n        if(input1 \u003d\u003d username \u0026\u0026 password \u003d\u003d input2) {\r\n          finished \u003d true\r\n        }\r\n        else {\r\n          SpecialPrint.cprintln(\"Invalid Credentials. Try again.\", Console.RED)\r\n        }\r\n\r\n      } while(!finished)\r\n\r\n      finished \u003d false\r\n\r\n      LibraryPrompter.printWelcomeMessage()\r\n      LibraryPrompter.printHelp()\r\n      do {\r\n        SpecialPrint.cprint(\"\u003e: \", Console.CYAN)\r\n        var input \u003d StdIn.readLine()\r\n        \r\n        input match {\r\n          case \"help\" \u003d\u003e LibraryPrompter.printHelp()\r\n          case \"search\" \u003d\u003e searchMenu();\r\n          case \"quit\" \u003d\u003e finished \u003d true\r\n          case _ \u003d\u003e SpecialPrint.cprintln(\"Unexpected command: Type \\\"help\\\" to relist possible commands\", Console.YELLOW)\r\n        }\r\n\r\n      } while(!finished)\r\n\r\n      //Close the connection\r\n      connection.close()\r\n    }\r\n    SpecialPrint.cprintln(\"Closing program\", Console.BLUE)\r\n  }\r\n\r\n  def searchMenu(): Unit \u003d {\r\n    LibraryPrompter.printSearchMenuPrompt()\r\n    var finished \u003d false\r\n    do {\r\n        SpecialPrint.cprint(\"search_attribute\u003e: \", Console.MAGENTA)\r\n        var input \u003d StdIn.readLine()\r\n        \r\n        input match {\r\n          case \"title\" \u003d\u003e queryDB(\"title\")\r\n          case \"author\" \u003d\u003e queryDB(\"author\")\r\n          case \"genre\" \u003d\u003e queryDB(\"genre\")\r\n          case \"publisher\" \u003d\u003e queryDB(\"publisher\")\r\n          case \"help\" \u003d\u003e LibraryPrompter.printSearchMenuHelp()\r\n          case \"back\" \u003d\u003e finished \u003d true\r\n          case _ \u003d\u003e SpecialPrint.cprintln(\"Unexpected command: Type \\\"help\\\" to relist possible commands\", Console.YELLOW)\r\n        }\r\n\r\n      } while(!finished)\r\n\r\n  }\r\n\r\n  def queryDB(attr:String): Unit \u003d {\r\n    SpecialPrint.cprint(attr + \"\u003e: \", Console.MAGENTA)\r\n    var input \u003d StdIn.readLine()\r\n\r\n    try {\r\n      val sql \u003d \"SELECT * FROM books WHERE \" + attr + \" LIKE \\\"%\" + input + \"%\\\"\"\r\n\r\n      // create the statement, and run the select query\r\n      val statement \u003d connection.createStatement()\r\n      val resultSet \u003d statement.executeQuery(sql)\r\n\r\n      if(resultSet.next() \u003d\u003d false) {\r\n        SpecialPrint.cprintln(\"No results found.\", Console.YELLOW)\r\n      }\r\n      else {\r\n        do {\r\n          var title \u003d resultSet.getString(\"title\")\r\n          var author \u003d resultSet.getString(\"author\")\r\n          var genre \u003d resultSet.getString(\"genre\")\r\n          var pages \u003d resultSet.getString(\"pages\")\r\n          var publisher \u003d resultSet.getString(\"publisher\")\r\n\r\n          println(\"Title: \" + title + \" | Author: \" + author + \" | Genre: \" + genre + \" | Pages: \" + pages + \" | Publisher: \" + publisher)\r\n        } while ( resultSet.next() )\r\n      }\r\n\r\n    }\r\n    catch {\r\n      case a: SQLException \u003d\u003e try {connection.rollback()} catch {case e: SQLException \u003d\u003e e.printStackTrace()}\r\n    } \r\n\r\n  }\r\n\r\n  def queryForBooksTest(): Unit \u003d {\r\n    try {\r\n      val sql \u003d \"SELECT * FROM books LIMIT 10\"\r\n\r\n      // create the statement, and run the select query\r\n      val statement \u003d connection.createStatement()\r\n      val resultSet \u003d statement.executeQuery(sql)\r\n\r\n      while ( resultSet.next() ) {\r\n        var title \u003d resultSet.getString(\"title\")\r\n        var author \u003d resultSet.getString(\"author\")\r\n        var genre \u003d resultSet.getString(\"genre\")\r\n        var pages \u003d resultSet.getString(\"pages\")\r\n        var publisher \u003d resultSet.getString(\"publisher\")\r\n        println(\"Title: \" + title + \" | Author: \" + author + \" | Genre: \" + genre + \" | Pages: \" + pages + \" | Publisher: \" + publisher)\r\n      }\r\n    }\r\n    catch {\r\n      case a: SQLException \u003d\u003e try {connection.rollback()} catch {case e: SQLException \u003d\u003e e.printStackTrace()}\r\n    }\r\n  }\r\n\r\n  def test_mysql_connection(): Int \u003d {\r\n\r\n    try {// connect to the database named \"demodatabase\" on the localhost\r\n      \r\n      val driver \u003d \"com.mysql.cj.jdbc.Driver\"\r\n      // make the connection\r\n      Class.forName(driver).newInstance()\r\n      connection \u003d DriverManager.getConnection(\"jdbc:mysql://localhost:3306/demodatabase?\" + \"user\u003droot\u0026password\u003dRo93Jo98@\")\r\n      connection.setAutoCommit(false);\r\n\r\n      // create the statement, and run the select query\r\n      val statement \u003d connection.createStatement()\r\n      val resultSet \u003d statement.executeQuery(\"SELECT response FROM connection_test\")\r\n\r\n      while ( resultSet.next() ) {\r\n        val status \u003d resultSet.getString(\"response\")\r\n        SpecialPrint.cprintln(\"Database Connection status \u003d \" + status, Console.GREEN)\r\n      }\r\n\r\n      //Return a status code of 1 for successful connection\r\n      return 1;\r\n\r\n    } \r\n    catch {\r\n      case a: ClassNotFoundException \u003d\u003e a.printStackTrace(); print_failure_message(\"Connection status \u003d Failure\"); return 0;\r\n      case b: SQLException \u003d\u003e b.printStackTrace(); print_failure_message(\"Connection status \u003d Failure\"); return 0;\r\n      case c: SQLTimeoutException \u003d\u003e c.printStackTrace(); print_failure_message(\"Connection status \u003d Failure\"); return 0;\r\n    } \r\n    finally {\r\n      // Scala wants return statement within catch block\r\n    }\r\n    \r\n  }\r\n\r\n  def print_failure_message(error_message:String): Unit \u003d {\r\n    SpecialPrint.cprintln(error_message, Console.RED)\r\n  }\r\n\r\n  def load_books_to_db(): Unit \u003d {\r\n    var lineReader:BufferedReader \u003d null\r\n    try {\r\n      connection.setAutoCommit(false); // To run multiple statements\r\n      lineReader \u003d new BufferedReader(new FileReader(\"books_updated.csv\"))\r\n      lineReader.readLine() // Skip the first line w/ the list of column names\r\n\r\n      var lineText:String \u003d lineReader.readLine() // If the variable is null i.e. empty file then the queries wont run\r\n      val sql \u003d \"INSERT INTO books (title, author, genre, pages, publisher) VALUES (?, ?, ?, ?, ?)\"\r\n      var statement:PreparedStatement \u003d connection.prepareStatement(sql)\r\n\r\n      println(\"Attempting queuries...\")\r\n\r\n      val batchSize \u003d 20\r\n      var count \u003d 0\r\n\r\n      while(lineText !\u003d null) {\r\n        var data \u003d lineText.split(\",\")\r\n        lineText \u003d lineReader.readLine()\r\n        var title \u003d data(0)\r\n        var author \u003d data(1)\r\n        var genre \u003d data(2)\r\n        var pages \u003d data(3).toInt\r\n        var publisher \u003d data(4)\r\n\r\n        //Build statement (query)\r\n        statement.setString(1, title)\r\n        statement.setString(2, author)\r\n        statement.setString(3, genre)\r\n        statement.setInt(4, pages)\r\n        statement.setString(5, publisher)\r\n\r\n        statement.addBatch()\r\n        if(count % batchSize \u003d\u003d 0) {\r\n          statement.executeBatch()\r\n        }\r\n        \r\n      }\r\n\r\n      lineReader.close()\r\n\r\n      // execute the remaining queries\r\n      statement.executeBatch();\r\n \r\n      connection.commit();\r\n\r\n    }\r\n    catch {\r\n      case a: FileNotFoundException \u003d\u003e a.printStackTrace(); print_failure_message(\"Requested file not found.\");\r\n      case b: SQLException \u003d\u003e try {connection.rollback()} catch {case e: SQLException \u003d\u003e e.printStackTrace()}\r\n    }\r\n\r\n  }\r\n\r\n}// EOF"
    }
  }
}
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.notify(GenericEndpoint.java:152)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleNotification(RemoteEndpoint.java:220)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:187)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	... 11 more
Caused by: java.nio.file.NoSuchFileException: c:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo\src\main\scala\HelloWorld.scala
	at sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:79)
	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:97)
	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:102)
	at sun.nio.fs.WindowsFileSystemProvider.newByteChannel(WindowsFileSystemProvider.java:230)
	at java.nio.file.Files.newByteChannel(Files.java:361)
	at java.nio.file.Files.newByteChannel(Files.java:407)
	at java.nio.file.Files.readAllBytes(Files.java:3152)
	at scala.meta.internal.io.PlatformFileIO$.slurp(PlatformFileIO.scala:45)
	at scala.meta.internal.io.FileIO$.slurp(FileIO.scala:24)
	at scala.meta.internal.metals.MetalsLanguageServer.didOpen(MetalsLanguageServer.scala:973)
	... 16 more

Oct 20, 2021 12:53:36 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: java.util.NoSuchElementException: BatchedFunction
java.util.concurrent.CompletionException: java.util.NoSuchElementException: BatchedFunction
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:661)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at scala.meta.internal.metals.CancelTokens$.$anonfun$future$1(CancelTokens.scala:40)
	at scala.meta.internal.metals.CancelTokens$.$anonfun$future$1$adapted(CancelTokens.scala:38)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.util.NoSuchElementException: BatchedFunction
	at scala.meta.internal.metals.BatchedFunction.<init>(BatchedFunction.scala:59)
	at scala.meta.internal.metals.MetalsLanguageServer.<init>(MetalsLanguageServer.scala:193)
	at scala.meta.metals.Main$.main(Main.scala:51)
	at scala.meta.metals.Main.main(Main.scala)

[0m2021.10.20 12:53:36 ERROR text document: file:///c:/Users/William%20Terry/Desktop/BigDataOctober2021/ScalaDBDemo/src/main/scala/HelloWorld.scala
java.nio.file.NoSuchFileException: c:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo\src\main\scala\HelloWorld.scala
	at sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:79)
	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:97)
	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:102)
	at sun.nio.fs.WindowsFileSystemProvider.newByteChannel(WindowsFileSystemProvider.java:230)
	at java.nio.file.Files.newByteChannel(Files.java:361)
	at java.nio.file.Files.newByteChannel(Files.java:407)
	at java.nio.file.Files.readAllBytes(Files.java:3152)
	at scala.meta.internal.io.PlatformFileIO$.slurp(PlatformFileIO.scala:45)
	at scala.meta.internal.io.FileIO$.slurp(FileIO.scala:24)
	at scala.meta.internal.metals.InteractiveSemanticdbs.$anonfun$textDocument$2(InteractiveSemanticdbs.scala:85)
	at scala.Option.getOrElse(Option.scala:189)
	at scala.meta.internal.metals.InteractiveSemanticdbs.$anonfun$textDocument$1(InteractiveSemanticdbs.scala:85)
	at java.util.HashMap.compute(HashMap.java:1197)
	at java.util.Collections$SynchronizedMap.compute(Collections.java:2684)
	at scala.meta.internal.metals.InteractiveSemanticdbs.textDocument(InteractiveSemanticdbs.scala:84)
	at scala.meta.internal.metals.InteractiveSemanticdbs.textDocument(InteractiveSemanticdbs.scala:58)
	at scala.meta.internal.metals.AggregateSemanticdbs.loop$1(AggregateSemanticdbs.scala:30)
	at scala.meta.internal.metals.AggregateSemanticdbs.textDocument(AggregateSemanticdbs.scala:36)
	at scala.meta.internal.metals.CodeLensProvider.findLenses(CodeLensProvider.scala:22)
	at scala.meta.internal.metals.MetalsLanguageServer.$anonfun$codeLens$2(MetalsLanguageServer.scala:1531)
	at scala.meta.internal.metals.TimerProvider.timedThunk(TimerProvider.scala:25)
	at scala.meta.internal.metals.MetalsLanguageServer.$anonfun$codeLens$1(MetalsLanguageServer.scala:1529)
	at scala.meta.internal.metals.CancelTokens$.$anonfun$apply$2(CancelTokens.scala:26)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
	at scala.util.Success.$anonfun$map$1(Try.scala:255)
	at scala.util.Success.map(Try.scala:213)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[0m
[0m2021.10.20 12:53:35 INFO  skipping build import with status 'Installed'[0m
Starting the bsp launcher for bloop...
Opening a bsp server connection with 'bsp --protocol tcp --port 29835'...
Waiting for the bsp connection to come up...
[0m[32m[D][0m Waiting for a connection at 127.0.0.1:29835...
The server is listening for incoming connections at tcp://127.0.0.1:29835...
Starting thread that pumps stdin and redirects it to the bsp server...
Starting thread that pumps server stdout and redirects it to the client stdout...
[0m2021.10.20 12:53:37 INFO  tracing is disabled for protocol BSP, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\bsp.trace.json[0m
[0m2021.10.20 12:53:37 INFO  Attempting to connect to the build server...[0m
Starting the bsp launcher for bloop...
Opening a bsp server connection with 'bsp --protocol tcp --port 25040'...
Waiting for the bsp connection to come up...
[0m[32m[D][0m Waiting for a connection at 127.0.0.1:25040...
The server is listening for incoming connections at tcp://127.0.0.1:25040...
Starting thread that pumps stdin and redirects it to the bsp server...
Starting thread that pumps server stdout and redirects it to the client stdout...
[0m2021.10.20 12:53:37 INFO  tracing is disabled for protocol BSP, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\bsp.trace.json[0m
[0m2021.10.20 12:53:37 INFO  time: Connected to build server in 1.35s[0m
[0m2021.10.20 12:53:37 INFO  Connected to Build server: Bloop v1.4.9[0m
[0m2021.10.20 12:53:42 INFO  time: indexed workspace in 4.68s[0m
[0m2021.10.20 12:53:44 INFO  no build target: using presentation compiler with only scala-library: 2.12.14[0m
[0m2021.10.20 12:53:45 INFO  compiling scaladbdemo (3 scala sources)[0m
[0m2021.10.20 12:53:47 INFO  time: compiled scaladbdemo in 2.87s[0m
[0m2021.10.20 12:57:23 INFO  compiling scaladbdemo (2 scala sources)[0m
[0m2021.10.20 12:57:23 INFO  time: compiled scaladbdemo in 0.49s[0m
[0m2021.10.20 13:02:09 INFO  Started: Metals version 0.10.7 in workspace 'C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)' for client Visual Studio Code 1.60.2.[0m
[0m2021.10.20 13:02:09 INFO  time: initialize in 0.55s[0m
[0m2021.10.20 13:02:14 INFO  Attempting to connect to the build server...[0m
[0m2021.10.20 13:02:14 WARN  no build target for: C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\src\main\scala\App.scala[0m
Starting the bsp launcher for bloop...
Opening a bsp server connection with 'bsp --protocol tcp --port 44911'...
Waiting for the bsp connection to come up...
[0m[32m[D][0m Loading 2 projects from 'c:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\.bloop'...
[0m[32m[D][0m Loading project from 'c:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\.bloop\scaladbdemo-test.json'
[0m[32m[D][0m Loading project from 'c:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\.bloop\scaladbdemo.json'
[0m[32m[D][0m Missing analysis file for project 'scaladbdemo-test'
[0m[32m[D][0m Loading previous analysis for 'scaladbdemo' from 'C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo\target\streams\compile\bloopAnalysisOut\_global\streams\inc_compile_2.11.zip'.
[0m[32m[D][0m Waiting for a connection at 127.0.0.1:44911...
The server is listening for incoming connections at tcp://127.0.0.1:44911...
Starting thread that pumps stdin and redirects it to the bsp server...
Starting thread that pumps server stdout and redirects it to the client stdout...
[0m2021.10.20 13:02:17 INFO  tracing is disabled for protocol BSP, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\bsp.trace.json[0m
[0m2021.10.20 13:02:17 INFO  Attempting to connect to the build server...[0m
Starting the bsp launcher for bloop...
Opening a bsp server connection with 'bsp --protocol tcp --port 29023'...
Waiting for the bsp connection to come up...
[0m[32m[D][0m Loading workspace settings from bloop.settings.json
[0m[32m[D][0m Waiting for a connection at 127.0.0.1:29023...
The server is listening for incoming connections at tcp://127.0.0.1:29023...
Starting thread that pumps stdin and redirects it to the bsp server...
Starting thread that pumps server stdout and redirects it to the client stdout...
[0m2021.10.20 13:02:19 INFO  tracing is disabled for protocol BSP, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\bsp.trace.json[0m
[0m2021.10.20 13:02:19 INFO  time: Connected to build server in 4.75s[0m
[0m2021.10.20 13:02:19 INFO  Connected to Build server: Bloop v1.4.9[0m
[0m2021.10.20 13:02:20 INFO  no build target: using presentation compiler with only scala-library: 2.12.15[0m
Oct 20, 2021 1:02:20 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation
WARNING: Unmatched cancel notification for request id 1
Oct 20, 2021 1:02:20 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation
WARNING: Unmatched cancel notification for request id 5
[0m2021.10.20 13:02:20 INFO  time: Imported build in 0.27s[0m
[0m2021.10.20 13:02:21 INFO  no build target: using presentation compiler with only scala-library: 2.12.15[0m
Oct 20, 2021 1:02:21 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation
WARNING: Unmatched cancel notification for request id 9
Oct 20, 2021 1:02:21 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation
WARNING: Unmatched cancel notification for request id 7
Oct 20, 2021 1:02:21 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation
WARNING: Unmatched cancel notification for request id 13
Oct 20, 2021 1:02:21 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation
WARNING: Unmatched cancel notification for request id 15
[0m2021.10.20 13:02:23 INFO  time: code lens generation in 7.26s[0m
[0m2021.10.20 13:02:23 INFO  time: code lens generation in 1.74s[0m
[0m2021.10.20 13:02:23 INFO  time: code lens generation in 1.84s[0m
[0m2021.10.20 13:02:27 INFO  time: indexed workspace in 6.25s[0m
[0m2021.10.20 13:02:27 WARN  no build target for: C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\src\main\scala\App.scala[0m
[0m2021.10.20 13:02:27 INFO  no build target: using presentation compiler with only scala-library: 2.12.14[0m
[0m2021.10.20 13:02:31 INFO  time: code lens generation in 1.25s[0m
[0m2021.10.20 13:06:33 WARN  no build target for: C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\src\main\scala\App.scala[0m
[0m2021.10.20 13:10:03 WARN  no build target for: C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\src\main\scala\App.scala[0m
[0m2021.10.20 13:10:04 INFO  compiling scaladbdemo (1 scala source)[0m
[0m2021.10.20 13:10:05 INFO  time: compiled scaladbdemo in 1.13s[0m
Oct 20, 2021 1:10:07 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: java.lang.ClassNotFoundException: Hello_World
java.util.concurrent.CompletionException: java.lang.ClassNotFoundException: Hello_World
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:661)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:29)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:26)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at scala.concurrent.BatchingExecutor$Batch.processBatch$1(BatchingExecutor.scala:67)
	at scala.concurrent.BatchingExecutor$Batch.$anonfun$run$1(BatchingExecutor.scala:82)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:59)
	at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:875)
	at scala.concurrent.BatchingExecutor.execute(BatchingExecutor.scala:110)
	at scala.concurrent.BatchingExecutor.execute$(BatchingExecutor.scala:107)
	at scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:873)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
	at scala.concurrent.Promise.complete(Promise.scala:53)
	at scala.concurrent.Promise.complete$(Promise.scala:52)
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:42)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ClassNotFoundException: Hello_World
	at scala.meta.internal.metals.debug.BuildTargetClassesFinder.$anonfun$findClassAndBuildTarget$1(BuildTargetClassesFinder.scala:107)
	at scala.Option.fold(Option.scala:251)
	at scala.meta.internal.metals.debug.BuildTargetClassesFinder.findClassAndBuildTarget(BuildTargetClassesFinder.scala:108)
	at scala.meta.internal.metals.debug.BuildTargetClassesFinder.findMainClassAndItsBuildTarget(BuildTargetClassesFinder.scala:36)
	at scala.meta.internal.metals.debug.DebugProvider.$anonfun$resolveMainClassParams$1(DebugProvider.scala:328)
	at scala.meta.internal.metals.debug.DebugProvider$$anonfun$withRebuildRetry$1.$anonfun$applyOrElse$5(DebugProvider.scala:560)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	... 4 more

Oct 20, 2021 1:10:29 PM org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint notify
INFO: Unsupported notification method: $/setTrace
Oct 20, 2021 1:10:31 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
java.util.concurrent.CompletionException: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:661)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:29)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:26)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at scala.concurrent.BatchingExecutor$Batch.processBatch$1(BatchingExecutor.scala:67)
	at scala.concurrent.BatchingExecutor$Batch.$anonfun$run$1(BatchingExecutor.scala:82)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:59)
	at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:875)
	at scala.concurrent.BatchingExecutor.execute(BatchingExecutor.scala:110)
	at scala.concurrent.BatchingExecutor.execute$(BatchingExecutor.scala:107)
	at scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:873)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
	at scala.concurrent.Promise.complete(Promise.scala:53)
	at scala.concurrent.Promise.complete$(Promise.scala:52)
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:42)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at scala.meta.internal.metals.debug.WorkspaceErrorsException$.<clinit>(DebugProvider.scala)
	at scala.meta.internal.metals.debug.DebugProvider.$anonfun$resolveMainClassParams$2(DebugProvider.scala:332)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	... 4 more

Oct 20, 2021 1:14:40 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation
WARNING: Unmatched cancel notification for request id 368
Oct 20, 2021 1:14:44 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
java.util.concurrent.CompletionException: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:661)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:29)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:26)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at scala.concurrent.BatchingExecutor$Batch.processBatch$1(BatchingExecutor.scala:67)
	at scala.concurrent.BatchingExecutor$Batch.$anonfun$run$1(BatchingExecutor.scala:82)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:59)
	at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:875)
	at scala.concurrent.BatchingExecutor.execute(BatchingExecutor.scala:110)
	at scala.concurrent.BatchingExecutor.execute$(BatchingExecutor.scala:107)
	at scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:873)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
	at scala.concurrent.Promise.complete(Promise.scala:53)
	at scala.concurrent.Promise.complete$(Promise.scala:52)
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:42)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at scala.meta.internal.metals.debug.WorkspaceErrorsException$.<clinit>(DebugProvider.scala)
	at scala.meta.internal.metals.debug.DebugProvider.$anonfun$resolveMainClassParams$2(DebugProvider.scala:332)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	... 4 more

[0m2021.10.20 13:14:48 INFO  running 'C:\Program Files\Java\jdk1.8.0_301\bin\java -Djline.terminal=jline.UnsupportedTerminal -Dsbt.log.noformat=true -Dfile.encoding=UTF-8 -jar C:\Users\WILLIA~1\AppData\Local\Temp\metals8822836420726526150\sbt-launch.jar -Dbloop.export-jar-classifiers=sources bloopInstall'[0m
[0m2021.10.20 13:14:50 INFO  [info] welcome to sbt 1.5.5 (Oracle Corporation Java 1.8.0_301)[0m
[0m2021.10.20 13:14:51 INFO  [info] loading settings for project scaladbdemo-copy-2--build-build from metals.sbt ...[0m
[0m2021.10.20 13:14:51 INFO  [info] loading project definition from C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\project\project[0m
[0m2021.10.20 13:14:53 INFO  [info] loading settings for project scaladbdemo-copy-2--build from metals.sbt ...[0m
[0m2021.10.20 13:14:53 INFO  [info] loading project definition from C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\project[0m
[0m2021.10.20 13:14:56 INFO  [success] Generated .bloop\scaladbdemo-copy-2--build.json[0m
[0m2021.10.20 13:14:56 INFO  [success] Total time: 4 s, completed Oct 20, 2021 1:14:56 PM[0m
[0m2021.10.20 13:14:58 INFO  [info] loading settings for project scaladbdemo-copy-2- from build.sbt ...[0m
[0m2021.10.20 13:14:58 INFO  [info] set current project to HelloWorldApp (in build file:/C:/Users/William%20Terry/Desktop/BigDataOctober2021/ScalaDBDemo%20-%20Copy%20(2)/)[0m
[0m2021.10.20 13:15:05 INFO  [success] Generated .bloop\scaladbdemo-copy-2--test.json[0m
[0m2021.10.20 13:15:05 INFO  [success] Generated .bloop\scaladbdemo-copy-2-.json[0m
[0m2021.10.20 13:15:05 INFO  [warn] Removed stale C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\.bloop\scaladbdemo-test.json[0m
[0m2021.10.20 13:15:05 INFO  [warn] Removed stale C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\.bloop\scaladbdemo.json[0m
[0m2021.10.20 13:15:05 INFO  [success] Total time: 6 s, completed Oct 20, 2021 1:15:05 PM[0m
[0m2021.10.20 13:15:05 INFO  time: ran 'sbt bloopInstall' in 16s[0m
[0m2021.10.20 13:15:05 INFO  Disconnecting from Bloop session...[0m
[0m2021.10.20 13:15:05 INFO  Shut down connection with build server.[0m
[0m2021.10.20 13:15:05 INFO No more data in the server stdin, exiting...
 No more data in the server stdin, exiting...
Shut down connection with build server.No more data in the server stdin, exiting...[0m

No more data in the server stdin, exiting...
No more data in the client stdin, exiting...
No more data in the client stdin, exiting...
[0m2021.10.20 13:15:05 INFO  Attempting to connect to the build server...[0m
Starting the bsp launcher for bloop...
Opening a bsp server connection with 'bsp --protocol tcp --port 43598'...
Waiting for the bsp connection to come up...
[0m[32m[D][0m Loading workspace settings from bloop.settings.json
[0m[32m[D][0m Loading 2 projects from 'c:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\.bloop'...
[0m[32m[D][0m Loading project from 'c:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\.bloop\scaladbdemo-copy-2--test.json'
[0m[32m[D][0m Loading project from 'c:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\.bloop\scaladbdemo-copy-2-.json'
[0m[32m[D][0m Configured SemanticDB in projects 'scaladbdemo-copy-2--test', 'scaladbdemo-copy-2-'
[0m[32m[D][0m Waiting for a connection at 127.0.0.1:43598...
The server is listening for incoming connections at tcp://127.0.0.1:43598...
Starting thread that pumps stdin and redirects it to the bsp server...
Starting thread that pumps server stdout and redirects it to the client stdout...
[0m2021.10.20 13:15:06 INFO  tracing is disabled for protocol BSP, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\bsp.trace.json[0m
[0m2021.10.20 13:15:06 INFO  Attempting to connect to the build server...[0m
Starting the bsp launcher for bloop...
Opening a bsp server connection with 'bsp --protocol tcp --port 19656'...
Waiting for the bsp connection to come up...
[0m[32m[D][0m Loading workspace settings from bloop.settings.json
[0m[32m[D][0m Waiting for a connection at 127.0.0.1:19656...
The server is listening for incoming connections at tcp://127.0.0.1:19656...
Starting thread that pumps stdin and redirects it to the bsp server...
Starting thread that pumps server stdout and redirects it to the client stdout...
[0m2021.10.20 13:15:05 INFO  tracing is disabled for protocol BSP, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\bsp.trace.json[0m
[0m2021.10.20 13:15:05 INFO  time: Connected to build server in 0.76s[0m
[0m2021.10.20 13:15:05 INFO  Connected to Build server: Bloop v1.4.9[0m
[0m2021.10.20 13:15:10 INFO  time: indexed workspace in 3.76s[0m
[0m2021.10.20 13:15:10 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:15:11 INFO  no build target: using presentation compiler with only scala-library: 2.12.14[0m
[0m2021.10.20 13:15:12 INFO  time: compiled scaladbdemo-copy-2- in 1.77s[0m
[0m2021.10.20 13:15:27 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 13:15:27 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 13:15:27 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 13:15:27 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:63320 .[0m
[0m2021.10.20 13:15:27 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 13:15:27 ERROR java.lang.ClassNotFoundException: com.postgresql.jdbc.Driver[0m
[0m2021.10.20 13:15:27 ERROR 	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)[0m
[0m2021.10.20 13:15:27 ERROR 	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)[0m
[0m2021.10.20 13:15:27 ERROR 	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:355)[0m
[0m2021.10.20 13:15:27 ERROR 	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)[0m
[0m2021.10.20 13:15:27 ERROR 	at java.lang.Class.forName0(Native Method)[0m
[0m2021.10.20 13:15:27 ERROR 	at java.lang.Class.forName(Class.java:264)[0m
[0m2021.10.20 13:15:27 ERROR 	at App$.main(App.scala:28)[0m
[0m2021.10.20 13:15:27 ERROR 	at App.main(App.scala)[0m
[0m2021.10.20 13:15:27 ERROR Exception in thread "main" java.lang.NullPointerException[0m
[0m2021.10.20 13:15:27 ERROR 	at App$.main(App.scala:42)[0m
[0m2021.10.20 13:15:27 ERROR 	at App.main(App.scala)[0m
[0m2021.10.20 13:15:27 INFO  Closing debug server tcp://0.0.0.0:63317[0m
[0m2021.10.20 13:15:27 INFO  Canceling debug proxy for [App][0m
Oct 20, 2021 1:16:53 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation
WARNING: Unmatched cancel notification for request id 386
Oct 20, 2021 1:16:59 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation
WARNING: Unmatched cancel notification for request id 405
[0m2021.10.20 13:17:25 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:17:25 INFO  time: compiled scaladbdemo-copy-2- in 0.5s[0m
[0m2021.10.20 13:17:27 INFO  skipping build import with status 'Installed'[0m
Oct 20, 2021 1:17:27 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
java.util.concurrent.CompletionException: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:661)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:29)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:26)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at scala.concurrent.BatchingExecutor$Batch.processBatch$1(BatchingExecutor.scala:67)
	at scala.concurrent.BatchingExecutor$Batch.$anonfun$run$1(BatchingExecutor.scala:82)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:59)
	at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:875)
	at scala.concurrent.BatchingExecutor.execute(BatchingExecutor.scala:110)
	at scala.concurrent.BatchingExecutor.execute$(BatchingExecutor.scala:107)
	at scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:873)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
	at scala.concurrent.Promise.complete(Promise.scala:53)
	at scala.concurrent.Promise.complete$(Promise.scala:52)
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:42)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at scala.meta.internal.metals.debug.WorkspaceErrorsException$.<clinit>(DebugProvider.scala)
	at scala.meta.internal.metals.debug.DebugProvider.$anonfun$resolveMainClassParams$2(DebugProvider.scala:332)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	... 4 more

[0m2021.10.20 13:28:13 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
Oct 20, 2021 1:28:13 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
java.util.concurrent.CompletionException: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:661)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:29)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:26)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at scala.concurrent.BatchingExecutor$Batch.processBatch$1(BatchingExecutor.scala:67)
	at scala.concurrent.BatchingExecutor$Batch.$anonfun$run$1(BatchingExecutor.scala:82)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:59)
	at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:875)
	at scala.concurrent.BatchingExecutor.execute(BatchingExecutor.scala:110)
	at scala.concurrent.BatchingExecutor.execute$(BatchingExecutor.scala:107)
	at scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:873)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
	at scala.concurrent.Promise.complete(Promise.scala:53)
	at scala.concurrent.Promise.complete$(Promise.scala:52)
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:42)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at scala.meta.internal.metals.debug.WorkspaceErrorsException$.<clinit>(DebugProvider.scala)
	at scala.meta.internal.metals.debug.DebugProvider.$anonfun$resolveMainClassParams$2(DebugProvider.scala:332)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	... 4 more

[0m2021.10.20 13:28:15 INFO  time: compiled scaladbdemo-copy-2- in 1.76s[0m
[0m2021.10.20 13:29:00 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:29:00 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 13m 54.21s)[0m
[0m2021.10.20 13:29:00 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:29:01 INFO  time: compiled scaladbdemo-copy-2- in 0.99s[0m
[0m2021.10.20 13:29:05 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 13:29:05 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 13:29:06 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 13:29:06 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:61918 .[0m
[0m2021.10.20 13:29:06 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 13:29:06 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 13:29:06 INFO  Closing debug server tcp://0.0.0.0:61915[0m
Oct 20, 2021 1:29:07 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation
WARNING: Unmatched cancel notification for request id 700
Oct 20, 2021 1:29:07 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation
WARNING: Unmatched cancel notification for request id 699
[0m2021.10.20 13:29:13 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:29:13 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 14m 7.07s)[0m
[0m2021.10.20 13:29:13 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:29:13 INFO  time: compiled scaladbdemo-copy-2- in 0.79s[0m
[0m2021.10.20 13:29:18 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 13:29:18 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 13:29:18 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 13:29:18 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:61930 .[0m
[0m2021.10.20 13:29:18 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 13:29:18 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 13:29:18 INFO  Closing debug server tcp://0.0.0.0:61927[0m
[0m2021.10.20 13:29:37 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:29:37 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 14m 31.488s)[0m
[0m2021.10.20 13:29:37 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:29:38 INFO  time: compiled scaladbdemo-copy-2- in 1.28s[0m
[0m2021.10.20 13:29:43 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 13:29:43 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 13:29:43 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 13:29:43 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:61942 .[0m
[0m2021.10.20 13:29:43 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 13:29:43 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 13:29:43 INFO  Closing debug server tcp://0.0.0.0:61939[0m
[0m2021.10.20 13:30:00 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:30:00 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 14m 54.818s)[0m
[0m2021.10.20 13:30:00 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:30:00 INFO  time: compiled scaladbdemo-copy-2- in 0.73s[0m
[0m2021.10.20 13:30:06 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 13:30:06 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 13:30:06 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 13:30:06 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:61954 .[0m
[0m2021.10.20 13:30:06 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 13:30:06 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 13:30:06 INFO  Closing debug server tcp://0.0.0.0:61951[0m
[0m2021.10.20 13:30:13 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:30:13 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 15m 7.705s)[0m
[0m2021.10.20 13:30:13 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:30:13 INFO  time: compiled scaladbdemo-copy-2- in 0.81s[0m
[0m2021.10.20 13:30:19 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 13:30:19 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 13:30:19 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 13:30:19 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:61966 .[0m
[0m2021.10.20 13:30:19 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 13:30:19 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 13:30:19 INFO  Closing debug server tcp://0.0.0.0:61963[0m
[0m2021.10.20 13:30:28 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:30:28 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 15m 22.533s)[0m
[0m2021.10.20 13:30:28 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:30:28 INFO  time: compiled scaladbdemo-copy-2- in 0.86s[0m
[0m2021.10.20 13:30:33 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 13:30:33 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 13:30:34 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 13:30:34 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:61978 .[0m
[0m2021.10.20 13:30:34 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 13:30:34 ERROR Exception in thread "main" org.postgresql.util.PSQLException: The authentication type 10 is not supported. Check that you have configured the pg_hba.conf file to include the client's IP address or subnet, and that it is using an authentication scheme supported by the driver.[0m
[0m2021.10.20 13:30:34 ERROR 	at org.postgresql.core.v3.ConnectionFactoryImpl.doAuthentication(ConnectionFactoryImpl.java:610)[0m
[0m2021.10.20 13:30:34 ERROR 	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:203)[0m
[0m2021.10.20 13:30:34 ERROR 	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:65)[0m
[0m2021.10.20 13:30:34 ERROR 	at org.postgresql.jdbc2.AbstractJdbc2Connection.<init>(AbstractJdbc2Connection.java:149)[0m
[0m2021.10.20 13:30:34 ERROR 	at org.postgresql.jdbc3.AbstractJdbc3Connection.<init>(AbstractJdbc3Connection.java:35)[0m
[0m2021.10.20 13:30:34 ERROR 	at org.postgresql.jdbc3g.AbstractJdbc3gConnection.<init>(AbstractJdbc3gConnection.java:22)[0m
[0m2021.10.20 13:30:34 ERROR 	at org.postgresql.jdbc4.AbstractJdbc4Connection.<init>(AbstractJdbc4Connection.java:47)[0m
[0m2021.10.20 13:30:34 ERROR 	at org.postgresql.jdbc42.AbstractJdbc42Connection.<init>(AbstractJdbc42Connection.java:21)[0m
[0m2021.10.20 13:30:34 ERROR 	at org.postgresql.jdbc42.Jdbc42Connection.<init>(Jdbc42Connection.java:28)[0m
[0m2021.10.20 13:30:34 ERROR 	at org.postgresql.Driver.makeConnection(Driver.java:415)[0m
[0m2021.10.20 13:30:34 ERROR 	at org.postgresql.Driver.connect(Driver.java:282)[0m
[0m2021.10.20 13:30:34 ERROR 	at java.sql.DriverManager.getConnection(DriverManager.java:664)[0m
[0m2021.10.20 13:30:34 ERROR 	at java.sql.DriverManager.getConnection(DriverManager.java:247)[0m
[0m2021.10.20 13:30:34 ERROR 	at App$.main(App.scala:30)[0m
[0m2021.10.20 13:30:34 ERROR 	at App.main(App.scala)[0m
[0m2021.10.20 13:30:34 INFO  Closing debug server tcp://0.0.0.0:61975[0m
[0m2021.10.20 13:30:34 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 13:31:41 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:31:41 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 16m 35.255s)[0m
[0m2021.10.20 13:31:41 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:31:41 INFO  time: compiled scaladbdemo-copy-2- in 0.22s[0m
[0m2021.10.20 13:31:43 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:31:43 INFO  time: compiled scaladbdemo-copy-2- in 0.21s[0m
Oct 20, 2021 1:31:45 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: scala.meta.internal.metals.MetalsBspException: BSP connection failed in the attempt to get: DebugSessionAddress.  Compilation not successful
java.util.concurrent.CompletionException: scala.meta.internal.metals.MetalsBspException: BSP connection failed in the attempt to get: DebugSessionAddress.  Compilation not successful
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:661)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:29)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:26)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at scala.concurrent.BatchingExecutor$Batch.processBatch$1(BatchingExecutor.scala:67)
	at scala.concurrent.BatchingExecutor$Batch.$anonfun$run$1(BatchingExecutor.scala:82)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:59)
	at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:875)
	at scala.concurrent.BatchingExecutor.execute(BatchingExecutor.scala:110)
	at scala.concurrent.BatchingExecutor.execute$(BatchingExecutor.scala:107)
	at scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:873)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
	at scala.concurrent.Promise.complete(Promise.scala:53)
	at scala.concurrent.Promise.complete$(Promise.scala:52)
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: scala.meta.internal.metals.MetalsBspException: BSP connection failed in the attempt to get: DebugSessionAddress.  Compilation not successful
	at scala.meta.internal.metals.BuildServerConnection$$anonfun$1.applyOrElse(BuildServerConnection.scala:237)
	at scala.meta.internal.metals.BuildServerConnection$$anonfun$1.applyOrElse(BuildServerConnection.scala:229)
	at scala.concurrent.Future.$anonfun$recoverWith$1(Future.scala:417)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	... 4 more

[0m2021.10.20 13:42:50 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
Oct 20, 2021 1:42:50 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
java.util.concurrent.CompletionException: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:661)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:29)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:26)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at scala.concurrent.BatchingExecutor$Batch.processBatch$1(BatchingExecutor.scala:67)
	at scala.concurrent.BatchingExecutor$Batch.$anonfun$run$1(BatchingExecutor.scala:82)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:59)
	at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:875)
	at scala.concurrent.BatchingExecutor.execute(BatchingExecutor.scala:110)
	at scala.concurrent.BatchingExecutor.execute$(BatchingExecutor.scala:107)
	at scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:873)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
	at scala.concurrent.Promise.complete(Promise.scala:53)
	at scala.concurrent.Promise.complete$(Promise.scala:52)
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:42)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at scala.meta.internal.metals.debug.WorkspaceErrorsException$.<clinit>(DebugProvider.scala)
	at scala.meta.internal.metals.debug.DebugProvider.$anonfun$resolveMainClassParams$2(DebugProvider.scala:332)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	... 4 more

[0m2021.10.20 13:42:52 INFO  time: compiled scaladbdemo-copy-2- in 1.39s[0m
[0m2021.10.20 13:45:32 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:45:32 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 30m 26.941s)[0m
[0m2021.10.20 13:45:32 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:45:32 INFO  time: compiled scaladbdemo-copy-2- in 0.83s[0m
[0m2021.10.20 13:45:38 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 13:45:38 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 13:45:38 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 13:45:38 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:62157 .[0m
[0m2021.10.20 13:45:38 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 13:45:38 ERROR Exception in thread "main" java.sql.SQLException: No suitable driver found for jdbc:mysql://localhost:3306/test?[0m
[0m2021.10.20 13:45:38 ERROR 	at java.sql.DriverManager.getConnection(DriverManager.java:689)[0m
[0m2021.10.20 13:45:38 ERROR 	at java.sql.DriverManager.getConnection(DriverManager.java:247)[0m
[0m2021.10.20 13:45:38 ERROR 	at App$.main(App.scala:26)[0m
[0m2021.10.20 13:45:38 ERROR 	at App.main(App.scala)[0m
[0m2021.10.20 13:45:38 INFO  Closing debug server tcp://0.0.0.0:62154[0m
[0m2021.10.20 13:45:38 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 13:45:52 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:45:52 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 30m 46.461s)[0m
[0m2021.10.20 13:45:52 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:45:52 INFO  time: compiled scaladbdemo-copy-2- in 0.79s[0m
[0m2021.10.20 13:45:57 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 13:45:57 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 13:45:58 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 13:45:58 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:62169 .[0m
[0m2021.10.20 13:45:58 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 13:45:58 ERROR Exception in thread "main" java.sql.SQLException: No suitable driver found for jdbc:mysql://localhost:3306/test?[0m
[0m2021.10.20 13:45:58 ERROR 	at java.sql.DriverManager.getConnection(DriverManager.java:689)[0m
[0m2021.10.20 13:45:58 ERROR 	at java.sql.DriverManager.getConnection(DriverManager.java:247)[0m
[0m2021.10.20 13:45:58 ERROR 	at App$.main(App.scala:26)[0m
[0m2021.10.20 13:45:58 ERROR 	at App.main(App.scala)[0m
[0m2021.10.20 13:45:58 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 13:45:58 INFO  Closing debug server tcp://0.0.0.0:62166[0m
Oct 20, 2021 1:47:13 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation
WARNING: Unmatched cancel notification for request id 1185
[0m2021.10.20 13:47:27 INFO  running 'C:\Program Files\Java\jdk1.8.0_301\bin\java -Djline.terminal=jline.UnsupportedTerminal -Dsbt.log.noformat=true -Dfile.encoding=UTF-8 -jar C:\Users\WILLIA~1\AppData\Local\Temp\metals2610127410887518295\sbt-launch.jar -Dbloop.export-jar-classifiers=sources bloopInstall'[0m
[0m2021.10.20 13:47:28 INFO  [info] welcome to sbt 1.5.5 (Oracle Corporation Java 1.8.0_301)[0m
[0m2021.10.20 13:47:29 INFO  [info] loading settings for project scaladbdemo-copy-2--build-build from metals.sbt ...[0m
[0m2021.10.20 13:47:30 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 13:47:29 INFO  [info] loading project definition from C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\project\project[0m
[0m2021.10.20 13:47:31 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 13:47:31 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 13:47:31 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:62433 .[0m
[0m2021.10.20 13:47:31 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 13:47:31 ERROR Exception in thread "main" java.sql.SQLException: No suitable driver found for jdbc:mysql://localhost:3306/test?[0m
[0m2021.10.20 13:47:31 ERROR 	at java.sql.DriverManager.getConnection(DriverManager.java:689)[0m
[0m2021.10.20 13:47:31 ERROR 	at java.sql.DriverManager.getConnection(DriverManager.java:247)[0m
[0m2021.10.20 13:47:31 ERROR 	at App$.main(App.scala:26)[0m
[0m2021.10.20 13:47:31 ERROR 	at App.main(App.scala)[0m
[0m2021.10.20 13:47:31 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 13:47:31 INFO  Closing debug server tcp://0.0.0.0:62430[0m
[0m2021.10.20 13:47:32 INFO  [info] loading settings for project scaladbdemo-copy-2--build from metals.sbt ...[0m
[0m2021.10.20 13:47:32 INFO  [info] loading project definition from C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\project[0m
[0m2021.10.20 13:47:35 INFO  [success] Generated .bloop\scaladbdemo-copy-2--build.json[0m
[0m2021.10.20 13:47:35 INFO  [success] Total time: 3 s, completed Oct 20, 2021 1:47:35 PM[0m
[0m2021.10.20 13:47:37 INFO  [info] loading settings for project scaladbdemo-copy-2- from build.sbt ...[0m
[0m2021.10.20 13:47:37 INFO  [info] set current project to HelloWorldApp (in build file:/C:/Users/William%20Terry/Desktop/BigDataOctober2021/ScalaDBDemo%20-%20Copy%20(2)/)[0m
[0m2021.10.20 13:47:43 INFO  [success] Generated .bloop\scaladbdemo-copy-2-.json[0m
[0m2021.10.20 13:47:43 INFO  [success] Generated .bloop\scaladbdemo-copy-2--test.json[0m
[0m2021.10.20 13:47:43 INFO  [success] Total time: 5 s, completed Oct 20, 2021 1:47:43 PM[0m
[0m2021.10.20 13:47:43 INFO  time: ran 'sbt bloopInstall' in 16s[0m
[0m2021.10.20 13:47:43 INFO  Disconnecting from Bloop session...[0m
[0m2021.10.20 13:47:43 INFO  Shut down connection with build server.[0m
No more data in the client stdin, exiting...
No more data in the server stdin, exiting...
[0m2021.10.20 13:47:43 INFO  Shut down connection with build server.[0m
No more data in the client stdin, exiting...
No more data in the server stdin, exiting...
[0m2021.10.20 13:47:43 INFO  Attempting to connect to the build server...[0m
Starting the bsp launcher for bloop...
Opening a bsp server connection with 'bsp --protocol tcp --port 24486'...
Waiting for the bsp connection to come up...
[0m[32m[D][0m Loading workspace settings from bloop.settings.json
[0m[32m[D][0m Loading 2 projects from 'c:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\.bloop'...
[0m[32m[D][0m Loading project from 'c:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\.bloop\scaladbdemo-copy-2--test.json'
[0m[32m[D][0m Loading project from 'c:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\.bloop\scaladbdemo-copy-2-.json'
[0m[32m[D][0m Configured SemanticDB in projects 'scaladbdemo-copy-2-', 'scaladbdemo-copy-2--test'
[0m[32m[D][0m Waiting for a connection at 127.0.0.1:24486...
The server is listening for incoming connections at tcp://127.0.0.1:24486...
Starting thread that pumps stdin and redirects it to the bsp server...
Starting thread that pumps server stdout and redirects it to the client stdout...
[0m2021.10.20 13:47:43 INFO  tracing is disabled for protocol BSP, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\bsp.trace.json[0m
[0m2021.10.20 13:47:43 INFO  Attempting to connect to the build server...[0m
Starting the bsp launcher for bloop...
Opening a bsp server connection with 'bsp --protocol tcp --port 42167'...
Waiting for the bsp connection to come up...
[0m[32m[D][0m Loading workspace settings from bloop.settings.json
[0m[32m[D][0m Waiting for a connection at 127.0.0.1:42167...
The server is listening for incoming connections at tcp://127.0.0.1:42167...
Starting thread that pumps stdin and redirects it to the bsp server...
Starting thread that pumps server stdout and redirects it to the client stdout...
[0m2021.10.20 13:47:44 INFO  tracing is disabled for protocol BSP, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\bsp.trace.json[0m
[0m2021.10.20 13:47:44 INFO  time: Connected to build server in 0.78s[0m
[0m2021.10.20 13:47:44 INFO  Connected to Build server: Bloop v1.4.9[0m
[0m2021.10.20 13:47:46 INFO  time: indexed workspace in 2.01s[0m
[0m2021.10.20 13:47:46 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:47:47 INFO  no build target: using presentation compiler with only scala-library: 2.12.14[0m
[0m2021.10.20 13:47:47 INFO  time: code lens generation in 1.07s[0m
[0m2021.10.20 13:47:48 INFO  time: compiled scaladbdemo-copy-2- in 1.53s[0m
Oct 20, 2021 1:47:53 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation
WARNING: Unmatched cancel notification for request id 1222
[0m2021.10.20 13:47:57 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 13:47:57 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 13:47:58 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 13:47:58 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:62458 .[0m
[0m2021.10.20 13:47:58 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 13:47:59 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 13:47:59 INFO  Closing debug server tcp://0.0.0.0:62455[0m
[0m2021.10.20 13:49:15 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:49:15 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 1m 31.223s)[0m
[0m2021.10.20 13:49:15 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:49:15 INFO  time: compiled scaladbdemo-copy-2- in 0.83s[0m
[0m2021.10.20 13:49:20 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 13:49:20 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 13:49:20 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 13:49:20 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:50821 .[0m
[0m2021.10.20 13:49:20 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 13:49:21 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 13:49:20 INFO  Closing debug server tcp://0.0.0.0:50818[0m
[0m2021.10.20 13:50:14 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:50:14 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 2m 30.723s)[0m
[0m2021.10.20 13:50:14 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:50:15 INFO  time: compiled scaladbdemo-copy-2- in 0.9s[0m
[0m2021.10.20 13:50:20 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 13:50:20 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 13:50:20 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 13:50:20 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:61340 .[0m
[0m2021.10.20 13:50:20 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 13:50:20 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 13:50:20 INFO  Closing debug server tcp://0.0.0.0:61337[0m
[0m2021.10.20 13:53:15 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:53:15 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 5m 31.579s)[0m
[0m2021.10.20 13:53:15 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:53:15 INFO  time: compiled scaladbdemo-copy-2- in 0.3s[0m
[0m2021.10.20 13:53:17 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:53:17 INFO  time: compiled scaladbdemo-copy-2- in 0.24s[0m
Oct 20, 2021 1:53:20 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: scala.meta.internal.metals.MetalsBspException: BSP connection failed in the attempt to get: DebugSessionAddress.  Compilation not successful
java.util.concurrent.CompletionException: scala.meta.internal.metals.MetalsBspException: BSP connection failed in the attempt to get: DebugSessionAddress.  Compilation not successful
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:661)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:29)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:26)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at scala.concurrent.BatchingExecutor$Batch.processBatch$1(BatchingExecutor.scala:67)
	at scala.concurrent.BatchingExecutor$Batch.$anonfun$run$1(BatchingExecutor.scala:82)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:59)
	at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:875)
	at scala.concurrent.BatchingExecutor.execute(BatchingExecutor.scala:110)
	at scala.concurrent.BatchingExecutor.execute$(BatchingExecutor.scala:107)
	at scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:873)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
	at scala.concurrent.Promise.complete(Promise.scala:53)
	at scala.concurrent.Promise.complete$(Promise.scala:52)
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: scala.meta.internal.metals.MetalsBspException: BSP connection failed in the attempt to get: DebugSessionAddress.  Compilation not successful
	at scala.meta.internal.metals.BuildServerConnection$$anonfun$1.applyOrElse(BuildServerConnection.scala:237)
	at scala.meta.internal.metals.BuildServerConnection$$anonfun$1.applyOrElse(BuildServerConnection.scala:229)
	at scala.concurrent.Future.$anonfun$recoverWith$1(Future.scala:417)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	... 4 more

[0m2021.10.20 13:54:53 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
Oct 20, 2021 1:54:53 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
java.util.concurrent.CompletionException: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:661)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:29)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:26)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at scala.concurrent.BatchingExecutor$Batch.processBatch$1(BatchingExecutor.scala:67)
	at scala.concurrent.BatchingExecutor$Batch.$anonfun$run$1(BatchingExecutor.scala:82)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:59)
	at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:875)
	at scala.concurrent.BatchingExecutor.execute(BatchingExecutor.scala:110)
	at scala.concurrent.BatchingExecutor.execute$(BatchingExecutor.scala:107)
	at scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:873)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
	at scala.concurrent.Promise.complete(Promise.scala:53)
	at scala.concurrent.Promise.complete$(Promise.scala:52)
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:42)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at scala.meta.internal.metals.debug.WorkspaceErrorsException$.<clinit>(DebugProvider.scala)
	at scala.meta.internal.metals.debug.DebugProvider.$anonfun$resolveMainClassParams$2(DebugProvider.scala:332)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	... 4 more

[0m2021.10.20 13:54:54 INFO  time: compiled scaladbdemo-copy-2- in 1.03s[0m
[0m2021.10.20 13:55:00 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 13:55:00 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 13:55:00 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 13:55:00 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:49776 .[0m
[0m2021.10.20 13:55:00 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 13:55:01 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 13:55:00 INFO  Closing debug server tcp://0.0.0.0:49773[0m
[0m2021.10.20 13:55:16 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:55:16 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 7m 33.122s)[0m
[0m2021.10.20 13:55:16 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:55:16 INFO  time: compiled scaladbdemo-copy-2- in 0.75s[0m
[0m2021.10.20 13:55:22 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 13:55:22 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 13:55:22 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 13:55:22 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:49789 .[0m
[0m2021.10.20 13:55:22 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 13:55:23 ERROR Exception in thread "main" java.sql.SQLException: Column Index out of range, 0 < 1.[0m
[0m2021.10.20 13:55:23 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:129)[0m
[0m2021.10.20 13:55:23 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97)[0m
[0m2021.10.20 13:55:23 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:89)[0m
[0m2021.10.20 13:55:23 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:63)[0m
[0m2021.10.20 13:55:23 ERROR 	at com.mysql.cj.jdbc.result.ResultSetImpl.checkColumnBounds(ResultSetImpl.java:489)[0m
[0m2021.10.20 13:55:23 ERROR 	at com.mysql.cj.jdbc.result.ResultSetImpl.getString(ResultSetImpl.java:871)[0m
[0m2021.10.20 13:55:23 ERROR 	at App$.main(App.scala:42)[0m
[0m2021.10.20 13:55:23 ERROR 	at App.main(App.scala)[0m
[0m2021.10.20 13:55:23 INFO  Closing debug server tcp://0.0.0.0:49786[0m
[0m2021.10.20 13:55:23 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 13:55:51 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:55:51 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 8m 7.696s)[0m
[0m2021.10.20 13:55:51 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:55:51 INFO  time: compiled scaladbdemo-copy-2- in 0.87s[0m
[0m2021.10.20 13:55:57 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 13:55:57 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 13:55:57 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 13:55:57 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:49803 .[0m
[0m2021.10.20 13:55:57 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 13:55:58 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 13:55:58 INFO  Closing debug server tcp://0.0.0.0:49800[0m
[0m2021.10.20 13:57:32 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:57:32 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 9m 48.944s)[0m
[0m2021.10.20 13:57:32 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:57:32 INFO  time: compiled scaladbdemo-copy-2- in 0.77s[0m
[0m2021.10.20 13:57:38 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 13:57:38 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 13:57:38 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 13:57:38 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:53539 .[0m
[0m2021.10.20 13:57:38 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 13:57:39 ERROR Exception in thread "main" java.sql.SQLException: Column Index out of range, 0 < 1.[0m
[0m2021.10.20 13:57:39 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:129)[0m
[0m2021.10.20 13:57:39 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97)[0m
[0m2021.10.20 13:57:39 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:89)[0m
[0m2021.10.20 13:57:39 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:63)[0m
[0m2021.10.20 13:57:39 ERROR 	at com.mysql.cj.jdbc.result.ResultSetImpl.checkColumnBounds(ResultSetImpl.java:489)[0m
[0m2021.10.20 13:57:39 ERROR 	at com.mysql.cj.jdbc.result.ResultSetImpl.getString(ResultSetImpl.java:871)[0m
[0m2021.10.20 13:57:39 ERROR 	at App$.main(App.scala:42)[0m
[0m2021.10.20 13:57:39 ERROR 	at App.main(App.scala)[0m
[0m2021.10.20 13:57:39 INFO  Closing debug server tcp://0.0.0.0:53536[0m
[0m2021.10.20 13:57:39 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 13:57:41 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:57:41 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 9m 57.473s)[0m
[0m2021.10.20 13:57:41 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 13:57:42 INFO  time: compiled scaladbdemo-copy-2- in 1.09s[0m
[0m2021.10.20 13:57:47 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 13:57:47 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 13:57:47 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 13:57:47 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:63157 .[0m
[0m2021.10.20 13:57:47 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 13:57:48 ERROR Exception in thread "main" java.sql.SQLException: Column Index out of range, 0 < 1.[0m
[0m2021.10.20 13:57:48 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:129)[0m
[0m2021.10.20 13:57:48 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97)[0m
[0m2021.10.20 13:57:48 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:89)[0m
[0m2021.10.20 13:57:48 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:63)[0m
[0m2021.10.20 13:57:48 ERROR 	at com.mysql.cj.jdbc.result.ResultSetImpl.checkColumnBounds(ResultSetImpl.java:489)[0m
[0m2021.10.20 13:57:48 ERROR 	at com.mysql.cj.jdbc.result.ResultSetImpl.getString(ResultSetImpl.java:871)[0m
[0m2021.10.20 13:57:48 ERROR 	at App$.main(App.scala:42)[0m
[0m2021.10.20 13:57:48 ERROR 	at App.main(App.scala)[0m
[0m2021.10.20 13:57:48 INFO  Closing debug server tcp://0.0.0.0:63154[0m
[0m2021.10.20 13:57:48 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 14:00:47 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:00:47 INFO  time: compiled scaladbdemo-copy-2- in 0.3s[0m
Oct 20, 2021 2:00:53 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
java.util.concurrent.CompletionException: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:661)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:29)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:26)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at scala.concurrent.BatchingExecutor$Batch.processBatch$1(BatchingExecutor.scala:67)
	at scala.concurrent.BatchingExecutor$Batch.$anonfun$run$1(BatchingExecutor.scala:82)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:59)
	at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:875)
	at scala.concurrent.BatchingExecutor.execute(BatchingExecutor.scala:110)
	at scala.concurrent.BatchingExecutor.execute$(BatchingExecutor.scala:107)
	at scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:873)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
	at scala.concurrent.Promise.complete(Promise.scala:53)
	at scala.concurrent.Promise.complete$(Promise.scala:52)
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:42)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at scala.meta.internal.metals.debug.WorkspaceErrorsException$.<clinit>(DebugProvider.scala)
	at scala.meta.internal.metals.debug.DebugProvider.$anonfun$resolveMainClassParams$2(DebugProvider.scala:332)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	... 4 more

[0m2021.10.20 14:01:06 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
Oct 20, 2021 2:01:06 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
java.util.concurrent.CompletionException: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:661)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:29)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:26)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at scala.concurrent.BatchingExecutor$Batch.processBatch$1(BatchingExecutor.scala:67)
	at scala.concurrent.BatchingExecutor$Batch.$anonfun$run$1(BatchingExecutor.scala:82)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:59)
	at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:875)
	at scala.concurrent.BatchingExecutor.execute(BatchingExecutor.scala:110)
	at scala.concurrent.BatchingExecutor.execute$(BatchingExecutor.scala:107)
	at scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:873)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
	at scala.concurrent.Promise.complete(Promise.scala:53)
	at scala.concurrent.Promise.complete$(Promise.scala:52)
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:42)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at scala.meta.internal.metals.debug.WorkspaceErrorsException$.<clinit>(DebugProvider.scala)
	at scala.meta.internal.metals.debug.DebugProvider.$anonfun$resolveMainClassParams$2(DebugProvider.scala:332)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	... 4 more

[0m2021.10.20 14:01:06 INFO  time: compiled scaladbdemo-copy-2- in 0.98s[0m
[0m2021.10.20 14:01:16 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:01:16 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 13m 32.773s)[0m
[0m2021.10.20 14:01:16 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:01:16 INFO  time: compiled scaladbdemo-copy-2- in 0.77s[0m
[0m2021.10.20 14:01:22 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 14:01:22 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 14:01:22 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 14:01:22 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:54756 .[0m
[0m2021.10.20 14:01:22 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 14:01:23 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 14:01:23 INFO  Closing debug server tcp://0.0.0.0:54753[0m
[0m2021.10.20 14:01:33 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:01:33 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 13m 50.021s)[0m
[0m2021.10.20 14:01:33 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:01:33 INFO  time: compiled scaladbdemo-copy-2- in 0.71s[0m
[0m2021.10.20 14:01:39 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 14:01:39 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 14:01:39 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 14:01:39 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:63557 .[0m
[0m2021.10.20 14:01:39 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 14:01:40 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 14:01:40 INFO  Closing debug server tcp://0.0.0.0:63554[0m
[0m2021.10.20 14:02:35 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:02:35 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 14m 51.962s)[0m
[0m2021.10.20 14:02:35 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:02:35 INFO  time: compiled scaladbdemo-copy-2- in 0.78s[0m
[0m2021.10.20 14:02:41 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 14:02:41 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 14:02:41 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 14:02:41 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:50386 .[0m
[0m2021.10.20 14:02:41 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 14:02:42 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 14:02:42 INFO  Closing debug server tcp://0.0.0.0:50383[0m
[0m2021.10.20 14:03:37 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:03:37 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 15m 53.478s)[0m
[0m2021.10.20 14:03:37 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:03:38 INFO  time: compiled scaladbdemo-copy-2- in 1.07s[0m
[0m2021.10.20 14:03:43 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 14:03:43 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 14:03:43 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 14:03:43 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:58397 .[0m
[0m2021.10.20 14:03:43 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 14:03:43 ERROR Exception in thread "main" java.sql.SQLException: Column Index out of range, 0 < 1.[0m
[0m2021.10.20 14:03:43 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:129)[0m
[0m2021.10.20 14:03:43 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97)[0m
[0m2021.10.20 14:03:43 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:89)[0m
[0m2021.10.20 14:03:43 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:63)[0m
[0m2021.10.20 14:03:43 ERROR 	at com.mysql.cj.jdbc.result.ResultSetImpl.checkColumnBounds(ResultSetImpl.java:489)[0m
[0m2021.10.20 14:03:43 ERROR 	at com.mysql.cj.jdbc.result.ResultSetImpl.getString(ResultSetImpl.java:871)[0m
[0m2021.10.20 14:03:43 ERROR 	at App$.main(App.scala:47)[0m
[0m2021.10.20 14:03:43 ERROR 	at App.main(App.scala)[0m
[0m2021.10.20 14:03:44 INFO  Closing debug server tcp://0.0.0.0:58394[0m
[0m2021.10.20 14:03:44 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 14:03:53 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:03:53 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 16m 9.887s)[0m
[0m2021.10.20 14:03:53 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:03:53 INFO  time: compiled scaladbdemo-copy-2- in 0.91s[0m
[0m2021.10.20 14:03:59 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 14:03:59 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 14:03:59 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 14:03:59 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:58410 .[0m
[0m2021.10.20 14:03:59 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 14:04:00 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 14:04:00 INFO  Closing debug server tcp://0.0.0.0:58407[0m
[0m2021.10.20 14:04:36 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:04:36 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 16m 52.847s)[0m
[0m2021.10.20 14:04:36 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:04:36 INFO  time: compiled scaladbdemo-copy-2- in 0.82s[0m
[0m2021.10.20 14:04:42 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 14:04:42 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 14:04:42 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 14:04:42 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:58426 .[0m
[0m2021.10.20 14:04:42 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 14:04:42 ERROR Exception in thread "main" java.sql.SQLException: Column Index out of range, 4 > 3. [0m
[0m2021.10.20 14:04:42 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:129)[0m
[0m2021.10.20 14:04:42 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97)[0m
[0m2021.10.20 14:04:42 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:89)[0m
[0m2021.10.20 14:04:42 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:63)[0m
[0m2021.10.20 14:04:42 ERROR 	at com.mysql.cj.jdbc.result.ResultSetImpl.checkColumnBounds(ResultSetImpl.java:494)[0m
[0m2021.10.20 14:04:42 ERROR 	at com.mysql.cj.jdbc.result.ResultSetImpl.getString(ResultSetImpl.java:871)[0m
[0m2021.10.20 14:04:42 ERROR 	at App$.main(App.scala:48)[0m
[0m2021.10.20 14:04:42 ERROR 	at App.main(App.scala)[0m
[0m2021.10.20 14:04:43 INFO  Closing debug server tcp://0.0.0.0:58423[0m
[0m2021.10.20 14:04:43 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 14:05:17 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:05:17 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 17m 33.413s)[0m
[0m2021.10.20 14:05:17 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:05:17 INFO  time: compiled scaladbdemo-copy-2- in 0.83s[0m
[0m2021.10.20 14:05:22 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 14:05:22 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 14:05:23 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 14:05:23 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:58451 .[0m
[0m2021.10.20 14:05:23 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 14:05:23 ERROR Exception in thread "main" java.sql.SQLException: Column Index out of range, 4 > 3. [0m
[0m2021.10.20 14:05:23 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:129)[0m
[0m2021.10.20 14:05:23 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97)[0m
[0m2021.10.20 14:05:23 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:89)[0m
[0m2021.10.20 14:05:23 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:63)[0m
[0m2021.10.20 14:05:23 ERROR 	at com.mysql.cj.jdbc.result.ResultSetImpl.checkColumnBounds(ResultSetImpl.java:494)[0m
[0m2021.10.20 14:05:23 ERROR 	at com.mysql.cj.jdbc.result.ResultSetImpl.getString(ResultSetImpl.java:871)[0m
[0m2021.10.20 14:05:23 ERROR 	at App$.main(App.scala:48)[0m
[0m2021.10.20 14:05:23 ERROR 	at App.main(App.scala)[0m
[0m2021.10.20 14:05:24 INFO  Closing debug server tcp://0.0.0.0:58448[0m
[0m2021.10.20 14:05:24 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 14:06:37 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:06:37 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 18m 53.909s)[0m
[0m2021.10.20 14:06:37 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:06:37 INFO  time: compiled scaladbdemo-copy-2- in 0.74s[0m
[0m2021.10.20 14:06:43 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 14:06:43 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 14:06:43 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 14:06:43 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:58465 .[0m
[0m2021.10.20 14:06:43 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 14:06:44 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 14:06:44 INFO  Closing debug server tcp://0.0.0.0:58462[0m
[0m2021.10.20 14:07:23 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:07:23 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 19m 39.969s)[0m
[0m2021.10.20 14:07:23 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:07:23 INFO  time: compiled scaladbdemo-copy-2- in 0.79s[0m
[0m2021.10.20 14:07:29 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 14:07:29 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 14:07:29 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 14:07:29 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:58478 .[0m
[0m2021.10.20 14:07:29 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 14:07:29 ERROR Exception in thread "main" java.sql.SQLException: Before start of result set[0m
[0m2021.10.20 14:07:29 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:129)[0m
[0m2021.10.20 14:07:29 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97)[0m
[0m2021.10.20 14:07:29 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:89)[0m
[0m2021.10.20 14:07:29 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:63)[0m
[0m2021.10.20 14:07:29 ERROR 	at com.mysql.cj.jdbc.result.ResultSetImpl.checkRowPos(ResultSetImpl.java:517)[0m
[0m2021.10.20 14:07:29 ERROR 	at com.mysql.cj.jdbc.result.ResultSetImpl.getString(ResultSetImpl.java:870)[0m
[0m2021.10.20 14:07:29 ERROR 	at App$.main(App.scala:48)[0m
[0m2021.10.20 14:07:29 ERROR 	at App.main(App.scala)[0m
[0m2021.10.20 14:07:30 INFO  Closing debug server tcp://0.0.0.0:58475[0m
[0m2021.10.20 14:07:30 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 14:08:03 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:08:03 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 20m 19.349s)[0m
[0m2021.10.20 14:08:03 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:08:04 INFO  time: compiled scaladbdemo-copy-2- in 1.04s[0m
[0m2021.10.20 14:08:08 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 14:08:08 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 14:08:09 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 14:08:09 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:57071 .[0m
[0m2021.10.20 14:08:09 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 14:08:10 ERROR Exception in thread "main" java.sql.SQLException: Before start of result set[0m
[0m2021.10.20 14:08:10 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:129)[0m
[0m2021.10.20 14:08:10 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97)[0m
[0m2021.10.20 14:08:10 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:89)[0m
[0m2021.10.20 14:08:10 ERROR 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:63)[0m
[0m2021.10.20 14:08:10 ERROR 	at com.mysql.cj.jdbc.result.ResultSetImpl.checkRowPos(ResultSetImpl.java:517)[0m
[0m2021.10.20 14:08:10 ERROR 	at com.mysql.cj.jdbc.result.ResultSetImpl.getString(ResultSetImpl.java:870)[0m
[0m2021.10.20 14:08:10 ERROR 	at App$.main(App.scala:48)[0m
[0m2021.10.20 14:08:10 ERROR 	at App.main(App.scala)[0m
[0m2021.10.20 14:08:10 INFO  Closing debug server tcp://0.0.0.0:57068[0m
[0m2021.10.20 14:08:10 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 14:14:19 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:14:19 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 26m 35.637s)[0m
[0m2021.10.20 14:14:19 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:14:21 INFO  time: compiled scaladbdemo-copy-2- in 2.23s[0m
[0m2021.10.20 14:14:26 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 14:14:26 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 14:14:27 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 14:14:27 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:57433 .[0m
[0m2021.10.20 14:14:27 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 14:14:28 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 14:14:28 INFO  Closing debug server tcp://0.0.0.0:57430[0m
[0m2021.10.20 14:14:57 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:14:57 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 27m 13.282s)[0m
[0m2021.10.20 14:14:57 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:14:59 INFO  time: compiled scaladbdemo-copy-2- in 2.1s[0m
[0m2021.10.20 14:15:04 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 14:15:04 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 14:15:04 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 14:15:04 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:57469 .[0m
[0m2021.10.20 14:15:04 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 14:15:06 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 14:15:06 INFO  Closing debug server tcp://0.0.0.0:57466[0m
Oct 20, 2021 2:16:14 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation
WARNING: Unmatched cancel notification for request id 2678
[0m2021.10.20 14:16:24 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:16:24 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 28m 40.381s)[0m
[0m2021.10.20 14:16:24 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:16:24 INFO  time: compiled scaladbdemo-copy-2- in 0.18s[0m
[0m2021.10.20 14:16:26 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:16:26 INFO  time: compiled scaladbdemo-copy-2- in 0.27s[0m
Oct 20, 2021 2:16:28 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: scala.meta.internal.metals.MetalsBspException: BSP connection failed in the attempt to get: DebugSessionAddress.  Compilation not successful
java.util.concurrent.CompletionException: scala.meta.internal.metals.MetalsBspException: BSP connection failed in the attempt to get: DebugSessionAddress.  Compilation not successful
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:661)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:29)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:26)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at scala.concurrent.BatchingExecutor$Batch.processBatch$1(BatchingExecutor.scala:67)
	at scala.concurrent.BatchingExecutor$Batch.$anonfun$run$1(BatchingExecutor.scala:82)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:59)
	at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:875)
	at scala.concurrent.BatchingExecutor.execute(BatchingExecutor.scala:110)
	at scala.concurrent.BatchingExecutor.execute$(BatchingExecutor.scala:107)
	at scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:873)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
	at scala.concurrent.Promise.complete(Promise.scala:53)
	at scala.concurrent.Promise.complete$(Promise.scala:52)
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: scala.meta.internal.metals.MetalsBspException: BSP connection failed in the attempt to get: DebugSessionAddress.  Compilation not successful
	at scala.meta.internal.metals.BuildServerConnection$$anonfun$1.applyOrElse(BuildServerConnection.scala:237)
	at scala.meta.internal.metals.BuildServerConnection$$anonfun$1.applyOrElse(BuildServerConnection.scala:229)
	at scala.concurrent.Future.$anonfun$recoverWith$1(Future.scala:417)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	... 4 more

Oct 20, 2021 2:16:53 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
java.util.concurrent.CompletionException: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:661)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:29)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:26)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at scala.concurrent.BatchingExecutor$Batch.processBatch$1(BatchingExecutor.scala:67)
	at scala.concurrent.BatchingExecutor$Batch.$anonfun$run$1(BatchingExecutor.scala:82)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:59)
	at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:875)
	at scala.concurrent.BatchingExecutor.execute(BatchingExecutor.scala:110)
	at scala.concurrent.BatchingExecutor.execute$(BatchingExecutor.scala:107)
	at scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:873)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
	at scala.concurrent.Promise.complete(Promise.scala:53)
	at scala.concurrent.Promise.complete$(Promise.scala:52)
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:42)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at scala.meta.internal.metals.debug.WorkspaceErrorsException$.<clinit>(DebugProvider.scala)
	at scala.meta.internal.metals.debug.DebugProvider.$anonfun$resolveMainClassParams$2(DebugProvider.scala:332)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	... 4 more

[0m2021.10.20 14:17:10 INFO  C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\src\main\scala\App.scala:48:86: stale bloop error: ')' expected but '.' found.
        println(resultSet.getString(1) +", " +resultSet.getString(2) + ", " resultSet.getString(3))
                                                                                     ^[0m
[0m2021.10.20 14:17:10 INFO  C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\src\main\scala\App.scala:48:99: stale bloop error: ';' expected but ')' found.
        println(resultSet.getString(1) +", " +resultSet.getString(2) + ", " resultSet.getString(3))
                                                                                                  ^[0m
[0m2021.10.20 14:17:10 INFO  C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\src\main\scala\App.scala:48:86: stale bloop error: ')' expected but '.' found.
        println(resultSet.getString(1) +", " +resultSet.getString(2) + ", " resultSet.getString(3))
                                                                                     ^[0m
[0m2021.10.20 14:17:10 INFO  C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\src\main\scala\App.scala:48:99: stale bloop error: ';' expected but ')' found.
        println(resultSet.getString(1) +", " +resultSet.getString(2) + ", " resultSet.getString(3))
                                                                                                  ^[0m
[0m2021.10.20 14:17:14 INFO  C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\src\main\scala\App.scala:48:86: stale bloop error: ')' expected but '.' found.
        println(resultSet.getString(1) +", " +resultSet.getString(2) + ", " resultSet.getString(3))
                                                                                     ^[0m
[0m2021.10.20 14:17:14 INFO  C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\src\main\scala\App.scala:48:99: stale bloop error: ';' expected but ')' found.
        println(resultSet.getString(1) +", " +resultSet.getString(2) + ", " resultSet.getString(3))
                                                                                                  ^[0m
[0m2021.10.20 14:17:14 INFO  C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\src\main\scala\App.scala:48:86: stale bloop error: ')' expected but '.' found.
        println(resultSet.getString(1) +", " +resultSet.getString(2) + ", " resultSet.getString(3))
                                                                                     ^[0m
[0m2021.10.20 14:17:14 INFO  C:\Users\William Terry\Desktop\BigDataOctober2021\ScalaDBDemo - Copy (2)\src\main\scala\App.scala:48:99: stale bloop error: ';' expected but ')' found.
        println(resultSet.getString(1) +", " +resultSet.getString(2) + ", " resultSet.getString(3))
                                                                                                  ^[0m
[0m2021.10.20 14:20:08 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
Oct 20, 2021 2:20:08 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
java.util.concurrent.CompletionException: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:661)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:29)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:26)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at scala.concurrent.BatchingExecutor$Batch.processBatch$1(BatchingExecutor.scala:67)
	at scala.concurrent.BatchingExecutor$Batch.$anonfun$run$1(BatchingExecutor.scala:82)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:59)
	at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:875)
	at scala.concurrent.BatchingExecutor.execute(BatchingExecutor.scala:110)
	at scala.concurrent.BatchingExecutor.execute$(BatchingExecutor.scala:107)
	at scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:873)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
	at scala.concurrent.Promise.complete(Promise.scala:53)
	at scala.concurrent.Promise.complete$(Promise.scala:52)
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:42)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: scala.meta.internal.metals.debug.WorkspaceErrorsException$: Cannot run class, since the workspace has errors.
	at scala.meta.internal.metals.debug.WorkspaceErrorsException$.<clinit>(DebugProvider.scala)
	at scala.meta.internal.metals.debug.DebugProvider.$anonfun$resolveMainClassParams$2(DebugProvider.scala:332)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	... 4 more

[0m2021.10.20 14:20:12 INFO  time: compiled scaladbdemo-copy-2- in 3.64s[0m
[0m2021.10.20 14:20:19 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:20:19 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 32m 35.558s)[0m
[0m2021.10.20 14:20:19 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:20:22 INFO  time: compiled scaladbdemo-copy-2- in 2.5s[0m
[0m2021.10.20 14:20:27 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 14:20:27 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 14:20:27 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 14:20:27 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:57512 .[0m
[0m2021.10.20 14:20:27 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 14:20:30 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 14:20:30 INFO  Closing debug server tcp://0.0.0.0:57509[0m
Oct 20, 2021 2:20:38 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation
WARNING: Unmatched cancel notification for request id 2925
Oct 20, 2021 2:20:38 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleCancellation
WARNING: Unmatched cancel notification for request id 2926
[0m2021.10.20 14:20:48 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:20:48 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 33m 4.295s)[0m
[0m2021.10.20 14:20:48 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:20:50 INFO  time: compiled scaladbdemo-copy-2- in 2.5s[0m
[0m2021.10.20 14:20:55 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 14:20:55 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 14:20:56 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 14:20:56 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:57525 .[0m
[0m2021.10.20 14:20:56 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 14:20:58 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 14:20:58 INFO  Closing debug server tcp://0.0.0.0:57522[0m
[0m2021.10.20 14:21:16 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:21:16 INFO  Deduplicating compilation of scaladbdemo-copy-2- from bsp client 'Metals 0.10.7' (since 33m 32.654s)[0m
[0m2021.10.20 14:21:16 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:21:19 INFO  time: compiled scaladbdemo-copy-2- in 2.54s[0m
[0m2021.10.20 14:21:24 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 14:21:24 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 14:21:24 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 14:21:24 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:57539 .[0m
[0m2021.10.20 14:21:24 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 14:21:27 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 14:21:27 INFO  Closing debug server tcp://0.0.0.0:57536[0m
[0m2021.10.20 14:23:53 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:23:56 INFO  time: compiled scaladbdemo-copy-2- in 2.54s[0m
[0m2021.10.20 14:31:03 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:31:06 INFO  time: compiled scaladbdemo-copy-2- in 2.87s[0m
[0m2021.10.20 14:31:13 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 14:31:13 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 14:31:14 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 14:31:14 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:62966 .[0m
[0m2021.10.20 14:31:14 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 14:31:16 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 14:31:16 INFO  Closing debug server tcp://0.0.0.0:62963[0m
[0m2021.10.20 14:31:39 INFO  compiling scaladbdemo-copy-2- (1 scala source)[0m
[0m2021.10.20 14:31:41 INFO  time: compiled scaladbdemo-copy-2- in 2.59s[0m
[0m2021.10.20 14:31:55 INFO  tracing is disabled for protocol dap-server, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-server.trace.json[0m
[0m2021.10.20 14:31:55 INFO  tracing is disabled for protocol dap-client, to enable tracing of incoming and outgoing JSON messages create an empty file at C:\Users\William Terry\AppData\Local\scalameta\metals\cache\dap-client.trace.json[0m
[0m2021.10.20 14:31:56 INFO  Starting debug proxy for [App][0m
[0m2021.10.20 14:31:56 INFO  Trying to attach to remote debuggee VM kubernetes.docker.internal:62981 .[0m
[0m2021.10.20 14:31:56 INFO  Attaching to debuggee VM succeeded.[0m
[0m2021.10.20 14:31:58 INFO  Canceling debug proxy for [App][0m
[0m2021.10.20 14:31:58 INFO  Closing debug server tcp://0.0.0.0:62978[0m
[0m2021.10.20 14:36:45 INFO  shutting down Metals[0m
[0m2021.10.20 14:36:45 INFO  Shut down connection with build server.[0m
No more data in the client stdin, exiting...
No more data in the server stdin, exiting...
[0m2021.10.20 14:36:45 INFO  Shut down connection with build server.[0m
No more data in the client stdin, exiting...
No more data in the server stdin, exiting...
